sdcflows/transform.py

```diff
@@ -325,7 +325,12 @@ def fit(
         for level in coeffs:
             wmat = grid_bspline_weights(target_reference, level)
             weights.append(wmat)
-            coeffs_data.append(level.get_fdata(dtype="float32").reshape(-1))
+
+            # Coefficients must be zero-padded because Scipy's BSpline pads the knots.
+            coeffs_data.append(np.pad(
+                level.get_fdata(dtype="float32"),
+                ((1, 1), (1, 1), (1, 1)),
+            ).reshape(-1))
```

Comment:
If we're not removing weights when fitting, the coefficients should match the size of the design matrix, so we should not need to pad them here, correct?